src/refresh.cc

```diff
@@ -693,12 +684,13 @@ refreshStats(StoreEntry * sentry)
     storeAppendPrintf(sentry, "\nRefresh pattern usage:\n\n");
     storeAppendPrintf(sentry, "  Used      \tChecks    \t%% Matches\tPattern\n");
     for (const RefreshPattern *R = Config.Refresh; R; R = R->next) {
-        storeAppendPrintf(sentry, "  %10" PRIu64 "\t%10" PRIu64 "\t%6.2f\t%s%s\n",
+        SBufStream os;
+        R->printPattern(os);
+        storeAppendPrintf(sentry, "  %10" PRIu64 "\t%10" PRIu64 "\t%6.2f\t" SQUIDSBUFPH "\n",
```

Comment:
sentry is a Packable. You should be able to avoid all the SBuf and SBufStream allocations by using a PackableStream instead of storeAppendPrintf().

Reply:
Not quote what I asked for. But the rest can be out of scope.